tests/integration/controller/mandatorymodules/mandatory_modules_controller_test.go

```diff
@@ -0,0 +1,151 @@
+package mandatory_test
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	compdescv2 "github.com/open-component-model/ocm/pkg/contexts/ocm/compdesc/versions/v2"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/pkg/status"
+	"github.com/kyma-project/lifecycle-manager/pkg/testutils/builder"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	. "github.com/kyma-project/lifecycle-manager/pkg/testutils"
+)
+
+var (
+	ErrWrongModulesStatus  = errors.New("modules status not correct")
+	ErrNoMandatoryManifest = errors.New("manifest for mandatory Module not found")
+)
+
+var _ = Describe("Mandatory Module Installation", Ordered, func() {
+	Context("Given Kyma with no Module and one mandatory ModuleTemplate on Control-Plane", func() {
+		kyma := NewTestKyma("no-module-kyma")
+		registerControlPlaneLifecycleForKyma(kyma)
+
+		It("Then Kyma CR should result in a ready state immediately as there are no modules", func() {
+			Eventually(KymaIsInState, Timeout, Interval).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, shared.StateReady).
+				Should(Succeed())
+		})
+
+		It("And Kyma CR should contain empty status.modules", func() {
+			Eventually(func() error {
+				createdKyma, err := GetKyma(ctx, controlPlaneClient, kyma.GetName(), kyma.GetNamespace())
+				if err != nil {
+					return err
+				}
+				if len(createdKyma.Status.Modules) != 0 {
+					return ErrWrongModulesStatus
+				}
+				return nil
+			}, Timeout, Interval).
+				Should(Succeed())
+		})
+
+		It("And Manifest CR for the Mandatory Module should be created with correct Owner Reference", func() {
+			Eventually(checkMandatoryManifestForKyma).
+				WithContext(ctx).
+				WithArguments(kyma, "kyma-project.io/template-operator").
+				Should(Succeed())
+		})
+	})
+
+	Context("Given Kyma with no Module and one mandatory ModuleTemplate on Control-Plane", func() {
+		kyma := NewTestKyma("skip-reconciliation-kyma")
+		kyma.Labels[shared.SkipReconcileLabel] = "true"
+		registerControlPlaneLifecycleForKyma(kyma)
+
+		It("When Kyma has 'skip-reconciliation' label, then no Mandatory Module Manifest should be created", func() {
+			Eventually(checkMandatoryManifestForKyma).
+				WithContext(ctx).
+				WithArguments(kyma, "kyma-project.io/template-operator").
+				Should(Equal(ErrNoMandatoryManifest))
+		})
+	})
+})
+
+func registerControlPlaneLifecycleForKyma(kyma *v1beta2.Kyma) {
```

Comment:
Please don't continue to introduce this function into different integration test packages, instead think about a solution to make those tests share this code.

Basically, we have `RegisterDefaultLifecycleForKyma` and `RegisterDefaultLifecycleForKymaWithoutTemplate` in `integration/controller/kyma`, those two functions can be extracted to `integration` package under a `helper.go` file, if you just refactor the parameters to accept `context.Context` and `client.Client`, then you can make those two function used by each test packages, of cause, change will affect many existing code if you apply those new function into existing code, but then you can create followup issue for that refactoring dedicated.

And for your function here, just need to reuse `RegisterDefaultLifecycleForKymaWithoutTemplate`, and additionally deploy your mandatory template.

A general guide here, if you can simply copy some existing code to achieve, always think upfront if you can refactor it to be reused.